src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,55 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int MULTIPLIER = 2;
```

Comment:
Change name of this constant so it communicates intent better:
```suggestion
    private static final int GROWTH_FACTOR= 2;
```